src/analysis/ab_test.py

```
@@ -0,0 +1,319 @@
+            results.append({
+                "segment":           seg,
+                "n_control":         len(ctrl),
+                "n_treatment":       len(trt),
+                "churn_rate_ctrl":   round(float(ctrl.mean()), 4),
+                "churn_rate_trt":    round(float(trt.mean()), 4),
+                "churn_reduction":   round(float(ctrl.mean() - trt.mean()), 4),
+                "p_value":           round(float(p_val), 6),
+                "significant":       bool(p_val < ALPHA),
+            })
+        except Exception:
+            continue
```

Ruff (0.15.13)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)